plugins/out_s3/s3.c

```diff
@@ -1129,7 +1201,12 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
                 return -1;
             }
         }
-        return FLB_RETRY;
+        if (retry_check) {
+            return FLB_RETRY;
+        }
+        else {
+            return FLB_ERROR;
+        }
```

Comment:
Seems like ret = -1 indicates error rather than FLB_ERROR which is of value 0

upload_data called by `send_upload_ request()`
https://github.com/fluent/fluent-bit/blob/760956f50cdc7eef3047df20d2299202a8c68594/plugins/out_s3/s3.c#L1660-L1664

Send_upload_request expects -1 for flb_error
https://github.com/fluent/fluent-bit/blob/760956f50cdc7eef3047df20d2299202a8c68594/plugins/out_s3/s3.c#L1721-L1762

Other place where send_upload_request is called
https://github.com/fluent/fluent-bit/blob/760956f50cdc7eef3047df20d2299202a8c68594/plugins/out_s3/s3.c#L2199-L2208

Could you look into whether -1 should be returned instead of FLB_ERROR?

Reply:
I think this is a bug. We need to unify the logic around one return value strategy.